test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -68,10 +68,10 @@ abstract class DirectMethodSenderBase : IDisposable
                 logger.LogInformation(e, $"Unauthorized exception caught with count {this.directMethodCount}");
                 return new Tuple<HttpStatusCode, ulong>(HttpStatusCode.Unauthorized, this.directMethodCount);
             }
-            catch (DeviceNotFoundException e)
+            catch (Exception e) when (e is System.Net.Http.HttpRequestException || e is IotHubException || e is IotHubCommunicationException)
             {
-                logger.LogInformation(e, $"DeviceNotFound exception caught with count {this.directMethodCount}");
-                return new Tuple<HttpStatusCode, ulong>(HttpStatusCode.NotFound, this.directMethodCount);
+                logger.LogInformation(e, $"Transient exception caught with count {this.directMethodCount}");
+                return new Tuple<HttpStatusCode, ulong>(HttpStatusCode.FailedDependency, this.directMethodCount);
```

Comment:
I have reordered these exceptions to catch most specific first. No other changes.